compilers/concrete-compiler/compiler/lib/Runtime/simulation.cpp

```diff
@@ -183,6 +189,29 @@ void sim_wop_pbs_crt(
 
 uint64_t sim_neg_lwe_u64(uint64_t plaintext) { return ~plaintext + 1; }
 
+uint64_t sim_add_lwe_u64(uint64_t lhs, uint64_t rhs, char *loc,
+                         bool is_signed) {
+  uint64_t result = lhs + rhs;
+  if (!is_signed && (lhs > UINT63_MAX - rhs || result > UINT63_MAX)) {
+    printf("WARNING at %s: overflow happened during addition in simulation\n",
+           loc);
+  }
+  return result;
+}
+
+uint64_t sim_mul_lwe_u64(uint64_t lhs, uint64_t rhs, char *loc,
+                         bool is_signed) {
+
+  uint64_t result = lhs * rhs;
+  if (!is_signed && rhs != 0 &&
```

Comment:
Same comment than above, should be checked even if  signed, checking against result is not necessary.